src/sharedData/visualization/components/VisualizationMapLayer.js

```diff
@@ -0,0 +1,381 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import bbox from '@turf/bbox';
+import _ from 'lodash/fp';
+import logger from 'terraso-client-shared/monitoring/logger';
+import { Box, Portal, Stack, Typography } from '@mui/material';
+
+import Layer from 'gis/components/Layer';
+import { useMap } from 'gis/components/Map';
+import mapboxgl from 'gis/mapbox';
+import { getLayerImage } from 'sharedData/visualization/visualizationMarkers';
+
+const DEFAULT_MARKER_OPACITY = 1;
+
+export const LAYER_TYPES = ['markers', 'polygons-outline', 'polygons-fill'];
+
+export const getLayerOpacity = (type, visualizationConfig) => {
+  const opacity =
+    visualizationConfig?.visualizeConfig?.opacity || DEFAULT_MARKER_OPACITY;
+  switch (type) {
+    case 'polygons-fill':
+      return opacity / 100;
+    default:
+      return DEFAULT_MARKER_OPACITY;
+  }
```

Comment:
suggestion: don't compute opacity if it's not going to be used.
```suggestion
  if (type !== 'polygons-fill') {
    return DEFAULT_MARKER_OPACITY;
  }
  const opacity =
    visualizationConfig?.visualizeConfig?.opacity || DEFAULT_MARKER_OPACITY;
  return opacity / 100;
  }
```